### DISCUSSION TITLE
Persist style tags in Astro integration

### DISCUSSION THREAD

**Discussion by Ocean-H1** [swup/swup#939] [Help]
### Describe your problem 🧐

Hi 👋 ,my project uses astro and swup and [oh-my-live2d](https://oml2d.com/).This library generates a `<style id="oml2d-global-style"></style>`based on the user's configuration and inserts it into the `<head>` through javascript after the first load. But after I switched pages(these pages have the same `<head>` ...),  this  `<style>` was lost.
This is my @swup/astro configuration :
![image](https://github.com/user-attachments/assets/ca170d12-75af-4082-ba54-7d2eaba4ee56)

### What is your question? 🙋‍♀️

I tried to copy the  `<style>`(dynamically added in the `<head>` through javascript) and write it directly in the `<head>`, and it worked as expected, but it's hard to maintain.I've tried [head-plugin](https://swup.js.org/plugins/head-plugin/), but it didn't solve the problem, perhaps because of the `dynamic addition`
So, how can I keep the `<style>` mentioned above

### URL to minimal reproduction 🔗

https://stackblitz.com/edit/github-kvqyqk

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by daun** [comment]
I don't think that's currently possible when using the Astro integration.

When [setting up swup manually](https://github.com/swup/astro?tab=readme-ov-file#control-over-the-initialization), you can enable the [persistAssets](https://swup.js.org/plugins/head-plugin/#persistassets) option of the Head Plugin.

Feel free to [open an issue](https://github.com/swup/astro/issues) on the Astro integration repo so we can keep track of this.
